README.md

```diff
@@ -1,12 +1,108 @@
 # preCICE example for FSI: 1D elastic tube
 
-For more information see [1]. Scenario taken from [2].
+## First Step: Choose a version
 
-**The quick path:** For building and running instructions refer to [this preCICE wiki page](https://github.com/precice/precice/wiki/Running-the-1D-elastic-tube-example).
+This tutorial comes in two distinct versions, one written in C++ and one in Python. The code for each are located in the folders `cxx` and `python` respectively.
 
-**The detailed path:** If you prefer, a tutorial can take you step by step through this case. See [this preCICE wiki page](https://github.com/precice/precice/wiki/Example-for-FSI:-1D-elastic-tube).
+Choose a version, then navigate down to the corresponding sections for [Python](#python-version) and [C++](#c-version) instructions.
 
+---
+## Python version
 
+This version is realized using the Python API for preCICE. Check [this entry in the preCICE wiki](https://github.com/precice/precice/wiki/1D-elastic-tube-using-the-Python-API) for more information on this example.
+
+### Requirements
+
+- [preCICE](https://github.com/precice/precice/wiki/Get-preCICE)
+
+- [preCICE Python Bindings](https://github.com/precice/precice/blob/develop/src/precice/bindings/python/README.md).
+
+**Note:** these requirements are specific to the Python variant of the code. If you wish to run the C++ version, you require different packages (see [this section](#c-version)).
+
+### How to run
+
+0. Clone this repository:
+    ```bash
+    $ git clone https://github.com/precice/elastictube1d.git
+    ```
+
+1. Navigate into the `python` directory and execute the `Allrun` script to run both solver participants directly:
+    ```bash
+    $ cd python/ && ./Allrun
+    ```
```

Comment:
Just tried this out and the simulation hangs on my system with the terminal showing:
```
Starting Fluid participant...
Starting Structure participant...

Waiting for participants to finish...
(you may run 'tail -f Fluid.log' in another terminal to check the progress)
Fluid participant finished. Waiting for Structure participant...
```
The Fluid.log contains:
```
(0) 15:09:24 [xml::XMLAttribute]:202 in readValueSpecific: ^[[31mERROR: ^[[0mString to Double error
```
The Structure.log contains:
```
(0) 15:09:23 [impl::SolverInterfaceImpl]:112 in configure: ^[[0mThis is preCICE version 1.6.1
(0) 15:09:23 [impl::SolverInterfaceImpl]:113 in configure: ^[[0mRevision info: v1.6.1-124-ged8e871c
(0) 15:09:23 [impl::SolverInterfaceImpl]:114 in configure: ^[[0mConfiguring preCICE with configuration: "./precice-config.xml"
(0) 15:09:23 [impl::SolverInterfaceImpl]:157 in configure: ^[[0mRun in coupling mode
(0) 15:09:23 [impl::SolverInterfaceImpl]:226 in initialize: ^[[0mSetting up master communication to coupling partner/s
```

Reply:
This is most likely due to the used locale.

Can you try executing `export LANG=en_US.UTF-8` prior to running the solvers?

Reply:
I tried `export LANG=en_US.UTF-8` before running the solvers using the `Allrun` script but I see the same behaviour as mentioned above.

Reply:
`Allrun` script worked on my system after I ran `export LC_NUMERIC=en_US.UTF-8`